accepted/0000-hub_retail.md

```diff
@@ -0,0 +1,213 @@
+
+Images + Image metadata
+-----------------------
+
+Current situation: an image is built on Buildhost, then pushed to SUMA Server htdocs directory. Then SUMA creates a db entry (including package list) and a pillar file.
+The pillar conains URL of image tarball, checksums and additional deployment details. It is used for image synchronization and deployment. These data are extracted 
+on Buildhost at the end of build phase. Some of the data can't be extracted later form the tarball.
+
+The SUMA API related to images is rather docker-centered. For Kiwi images it allows to list images and packages and delete an image.
+
+Use cases currently not covered:
+
+- Move image tarball to external HTTP server
+  - In large environments it makes sense to use one or more servers for distribution of images, it should not be the Hub itself.
+- Copy image metadata from one SUMA Server to another 
+  - Hub - copy to all servers
+  - Test in lab, then move the same image to production.
+- Adjust deployment details
+
+
+Proposed solution:
+
+- Implement API calls to
+  - Get Kiwi Image pillar
+  - Update Kiwi Image pillar
+  - Export Kiwi Image (complete metadata)
+  - Import Kiwi Image
+
+- Implement a tool that handles the use cases above
+
+
+HWTYPE configurations
+---------------------
+
+This is currently implemented as a formula assigned to a group with special name.
+
+FIXME: pick one alternative
+
+Alternative 1:
+Extend retail_yaml to create the groups and formula data via Hub and Peripheral Server API.
+This is relatively small change to the code.
+
+Alternative 2:
+Central creation of groups is covered in [Hub Users RFC](00061-hub-users.md). The salt states
+can be extended to handle also formula data.
+This approach is probably more flexible.
+
+Both alternatives will have to cope with authentization and Hub vs. Peripheral server user credentials.
+
+YAML configuration
+------------------
+
+FIXME: optional, related to Alternative 2 in previous section
+
+We have retail_yaml tool for initial configuration of Retail environment from YAML file.
+[Hub Users RFC](00061-hub-users.md) introduced similar concept, based on Salt.
+Maybe these parts could be merged.
+
+
+Combined SUMA and Branch server
+-------------------------------
+
+Branch services are configured by Salt formulas. Also PXE entries for each terminal are generated
+by Salt (currently via reactor).
+With combined SUMA and Branch server, these formulas should be applied on SUMA server itself.
+
+Terminals connect to SUMA directly, SUMA Proxy is not needed.
+
+
+FIXME: pick one alternative
+
+Alternative 1: Hub as a Salt Master for Combined SUMA and Branch server
+
+- this is already used for [Hub Users RFC](00061-hub-users.md)
+
+problems:
+- does not work for PXE entries and disconnected operation
+- mixed environment with Hub, Peripheral Servers with standalone Branch Serveres and 
+  Combined Peripheral Servers will have non-optimal configuration
+- does not cover the "Single server" setup
```

Comment:
How does the current use case with one single server is covered now? IIUC the branch server would be instaled in a different machine from suse manager server

Reply:
Currently there is no supported single server scenario. Supported scenario always includes branch server proxy.
I think that will remain the case for non-hub environment, unless we safely figure out how to enable single Uyuni server to manage itself (including applying salt states on itself).

Reply:
> enable single Uyuni server to manage itself (including applying salt states on itself).

What management features would be needed? Registering a Server onto itself is not an option at the moment.

But applying Salt states is very easy, we can use `salt-call --local` especially if it's a one-shot setup only.

Reply:
Talking about non-hub usage it would need to read formula pillars and apply states (formula states and pxe entry)

In hub environment, there is only issue about pxe entry state as the configuration of formulas can be done from hub.

Reply:
> Talking about non-hub usage it would need to read formula pillars and apply states (formula states and pxe entry)

OK, so IIUC in this scenario using `salt-call --local` could be sufficient. We would "only" need to allow formulas with forms to be applied to the Server itself optionally, correct?

> In hub environment, there is only issue about pxe entry state as the configuration of formulas can be done from hub.

IIUC that problem would be solved via https://github.com/uyuni-project/uyuni-rfc/pull/46, or is there something else?